pkg/virt-controller/watch/application_test.go

```diff
@@ -40,6 +46,38 @@ func newValidGetRequest() *http.Request {
 var _ = Describe("Application", func() {
 	var app VirtControllerApp = VirtControllerApp{}
 
+	Describe("Reinitialization conditions", func() {
+		table.DescribeTable("Re-trigger initialization", func(hasCdiAtInit bool, addCrd bool, removeCrd bool, expectReInit bool) {
+			app := VirtControllerApp{}
+
+			clusterConfig, _, crdInformer := testutils.NewFakeClusterConfig(&kubev1.ConfigMap{})
+			app.clusterConfig = clusterConfig
+			app.reInitChan = make(chan struct{})
+			app.reInitTriggered = false
+			app.reInitLock = &sync.Mutex{}
+			app.hasCdi = hasCdiAtInit
+
+			app.clusterConfig.SetConfigModifiedCallback(app.configModificationCallback)
+
+			if addCrd {
+				testutils.AddDataVolumeAPI(crdInformer)
+			} else if removeCrd {
+				testutils.RemoveDataVolumeAPI(crdInformer)
+			}
+
+			time.Sleep(3)
```

Comment:
please document why is sleep needed and why is it 3

Reply:
added a comment here in the code. The sleep is to account for the async callback that occurs to trigger the reInitialization.

Reply:
Are you sure that you need the wait? Can't you just remove the `default` in the `select` below and insteada add a timeout to the select?

Reply:
isn't that practically the same thing?

Reply:
oh, i see, it's slightly more efficient time wise

Reply:
The timeout is pretty small, it may or may not work predictable. If you wait on the channel with a timeout, then you can give it a timeout of e.g. 1 second, but if the test works it will finish immediately.